app/src/main/java/chat/rocket/android/wallet/BlockchainInterface.java

```diff
@@ -0,0 +1,269 @@
+     */
+    public List<TransactionViewModel> getTransactions(String userWalletAddress, List<String> transactionHashes) {
+        // Get web3 Transaction objects from the hashes
+        List<Transaction> transactions = new ArrayList<>();
+        try {
+            for (String tx : transactionHashes) {
+                transactions.add(web3.ethGetTransactionByHash(tx).send().getTransaction());
+            }
+        } catch (Exception ex) {
+            Timber.e(ex);
+            return new ArrayList<>();
+        }
+
+        // Map the Transaction objects to TransactionViewModels
+        List<TransactionViewModel> txModels = new ArrayList<>();
+        for (Transaction tx: transactions) {
+            Boolean sentFromUser = tx.getFrom().substring(2).equals(userWalletAddress);
+            txModels.add(new TransactionViewModel(tx.getHash(),
+                    Convert.fromWei(tx.getValue().toString(), Convert.Unit.ETHER),
+                    getTimeStamp(tx).longValue(),
+                    sentFromUser));
+        }
+        Collections.reverse(txModels);
+        return txModels;
+    }
+
+    /**
+     * Get the time the transaction was mined by getting the block it is in
+     * @param tx
+     * @return timestamp as BigInteger value
+     */
+    private BigInteger getTimeStamp(Transaction tx) {
+        try {
+            return web3.ethGetBlockByHash(tx.getBlockHash(), false).sendAsync().get().getBlock().getTimestamp();
+        } catch (Exception ex) {
+            return BigInteger.ZERO;
+        }
+    }
+
+    /**
+     * Check if a given wallet address has a corresponding wallet file stored on the device
+     */
+    public Boolean walletFileExists(String address, Context c) {
+        return getFilesByAddress(address, c).length > 0;
+    }
+
+    /**
+     * Get balance in Wei of a wallet
+     * @param address the address/public key of the wallet
+     * @return BigDecimal amount of Wei in the wallet or
+     *          -1 if the call failed
+     */
+    public BigDecimal getBalance(String address) {
+        try {
+            EthGetBalance ethGetBalance = this.web3
+                    .ethGetBalance(addAddressPrefix(address), DefaultBlockParameterName.LATEST)
+                    .sendAsync()
+                    .get();
+            return Convert.fromWei(ethGetBalance.getBalance().toString(), Convert.Unit.ETHER);
+        } catch(Exception e) {
+            Timber.d("Exception Caught: ethGetBalance failed!");
+            e.printStackTrace();
+        }
+        return BigDecimal.valueOf(-1);
+    }
+
+    /**
+     * Create a new account on the blockchain and save private key file to internal storage
+     * @param password user's password to make the account/wallet
+     * @param c Context of the app's current activity
+     * @return the public key of the account/wallet
+     */
+    public String createWallet(String password, Context c) {
```

Comment:
Is this function used?

Reply:
No, and I suppose it won't be used in the future. I'll remove it